### ISSUE TITLE
Fails to open a extremely large .sh file

### ISSUE THREAD

**Issue by mksinicus** [helix-editor/helix#6113]
### Summary

I tried to open a large (247.8 MB) `.sh` file with helix, which contains a tar archive at its EOF. Then the terminal freezed for about 40s and helix didn't start, leaving the message below:

```
munmap_chunk(): invalid pointer
Aborted (core dumped)
```

### Reproduction Steps

```sh
$ hx racket-8.8-v86_64-linux-cs.sh
munmap_chunk(): invalid pointer
Aborted (core dumped)
```

`racket-8.8-v86_64-linux-cs.sh` was downloaded from <https://download.racket-lang.org/installers/8.8/racket-8.8-x86_64-linux-cs.sh>

### Helix log

`RUST_BACKTRACE=1 hx -vv` prints the same thing as above.

Below are the only entries of the day. Seems irrelevant to me (?)

~/.cache/helix/helix.log 

```
2023-02-26T21:55:11.993 helix_loader [DEBUG] Located configuration folders: []
2023-02-26T21:55:12.004 helix_view::clipboard [DEBUG] Using xclip to interact with the system and selection (primary) clipboard
```

### Platform

Linux (Kubuntu 22.04)

### Terminal Emulator

Alacritty 0.11.0, as well as Konsole 21.12.3

### Helix Version

helix 22.12 (98a3d469)

**Comment by pascalkuthe** [comment]
likely caused by the TS grammar, please try removing the file extension and opening the file again so it's treated as a text file

**Comment by mksinicus** [comment]
> likely caused by the TS grammar, please try removing the file extension and opening the file again so it's treated as a text file

I agree with you. Removing the extension and the shebang line worked.
